### PR TITLE
Time format option for watches

### PR DESCRIPTION
**Pull request by WilliamMurdoch**
Allows watches to change their displayed time format from 24 hour time to 12 hour time.

### CODE REVIEW

**Comment by WilliamMurdoch** [comment]
!review

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 1]
```diff
@@ -161,13 +163,28 @@
...
 /obj/item/clothing/gloves/watch/verb/checktime()
 	set category = "Object"
 	set name = "Check Time"
 	set src in usr
 
 	if(wired && !clipped)
-		to_chat(usr, "You check your watch, spotting a digital collection of numbers reading '[worldtime2text()]'. Today's date is '[time2text(world.time, "Month DD")]. [game_year]'.")
+		if(use_military_format)
+			to_chat(usr, "You check your watch, spotting a digital collection of numbers reading '[worldtime2text()]'. Today's date is '[time2text(world.time, "Month DD")]. [game_year]'.")
+		else if(!use_military_format)
```
Comment (Amunak) [inline code/modules/clothing/gloves/miscellaneous.dm:182]: No additional "if" needed here.

```suggestion
		else
```

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 2]
```diff
@@ -132,6 +132,8 @@
 	body_parts_covered = null
 	fingerprint_chance = 100
 	var/flipped = 0
+	var/use_military_format = TRUE
+	var/modified_time = 0
```
Comment (Amunak) [inline code/modules/clothing/gloves/miscellaneous.dm:136]: This should not be an object-level variable - move it to where you are using it in `checktime`.
```suggestion
```

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 3]
```diff
@@ -161,13 +163,28 @@
...
+
 /obj/item/clothing/gloves/watch/verb/checktime()
 	set category = "Object"
 	set name = "Check Time"
 	set src in usr
 
 	if(wired && !clipped)
-		to_chat(usr, "You check your watch, spotting a digital collection of numbers reading '[worldtime2text()]'. Today's date is '[time2text(world.time, "Month DD")]. [game_year]'.")
+		if(use_military_format)
+			to_chat(usr, "You check your watch, spotting a digital collection of numbers reading '[worldtime2text()]'. Today's date is '[time2text(world.time, "Month DD")]. [game_year]'.")
```
Comment (Amunak) [inline code/modules/clothing/gloves/miscellaneous.dm:181]: I'd consider saving just the clock text to a separate variable, then using a single `to_chat` to output it for readability / ease of changing the message.

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 4]
```diff
@@ -161,13 +163,28 @@
...
+		else if(!use_military_format)
+			if(worldtime2hours() <= 12)
+				to_chat(usr, "You check your watch, spotting a digital collection of numbers reading '[worldtime2text()] AM'. Today's date is '[time2text(world.time, "Month DD")]. [game_year]'.")
+			else if(worldtime2hours() >= 12)
+				modified_time = time2text(world.time + roundstart_hour HOURS - 12 HOURS, "hh:mm")
```
Comment (Amunak) [inline code/modules/clothing/gloves/miscellaneous.dm:186]:
```suggestion
				var/modified_time = time2text(world.time + roundstart_hour HOURS - 12 HOURS, "hh:mm")
```

**Review by Amunak — CHANGES_REQUESTED** [review]

File: code/modules/clothing/gloves/miscellaneous.dm (hunk 3 above)
Comment (WilliamMurdoch) [reply code/modules/clothing/gloves/miscellaneous.dm:180]: I might go back and make that change. For now I'm going to commit the suggestions.

**Review by Amunak — APPROVED** [review]

File: code/modules/clothing/gloves/miscellaneous.dm [hunk 5]
```diff
@@ -161,13 +162,30 @@
...
 /obj/item/clothing/gloves/watch/verb/checktime()
 	set category = "Object"
 	set name = "Check Time"
 	set src in usr
 
+	var/modified_time
+	if(use_military_format)
+		modified_time = worldtime2text()
+	else
+		if(worldtime2hours() <= 12)
```
Comment (Hockaa) [inline code/modules/clothing/gloves/miscellaneous.dm:182]: Wouldn't this cause the watch to display 12 in the afternoon as 12 AM?

Comment (WilliamMurdoch) [reply code/modules/clothing/gloves/miscellaneous.dm:182]: That is a good point

Comment (Doxxmedearly) [reply code/modules/clothing/gloves/miscellaneous.dm:182]: Will approve if you update this. You can simply do < 12 instead of <= 12.

Comment (Doxxmedearly) [reply code/modules/clothing/gloves/miscellaneous.dm:182]: Though you may need to fiddle with the below else statement as well.

Comment (WilliamMurdoch) [reply code/modules/clothing/gloves/miscellaneous.dm:182]: Alright so I seemed to have made a mistake somewhere trying to add that commit. Now the branch this PR is on is out of date and it won't let me fetch origin.

**Comment by NonQueueingMatt** [comment]
Reopen when ready.

**Pull request closed without merging**